### ISSUE TITLE
Should HTTP headers in .test be in the same order CB provide them?

### ISSUE THREAD

**Issue by fgalan** [telefonicaid/fiware-orion#3473]
Based in cases like [this one](https://github.com/telefonicaid/fiware-orion/pull/3472/files/bf689df90741336365a6a8562f8216c9cd30311a#r276383701), it could happend that current implementation of accumulator-server.py doesn't dump the headers in the same order it gets them.

Should HTTP headers in .test be in the same order CB provide them? I'm not so sure:

* In the one hand, that way .test would be strictly precise (and strict precission uses to be good in regressions)
* In the other hand, HTTP standard specifies that the order is not important

Moreover, although the "problem" could be fixed in accumulator-server.py (although it seems it requires to use Python 3.7, see https://stackoverflow.com/questions/55751419/getting-request-headers-in-the-same-order-they-are-received-in-flask/55752367#55752367... no problem in a virtualenv, I guess) maybe libcurl introduces another point of uncertainty (libcurl literature should be checked).

**Comment by kzangeli** [comment]
We might have to sort the HTTP headers alphabetically for this to work on all nodes

**Comment by kzangeli** [comment]
I would **not** want to use the #SORT_START/END mechanism of the functional tests ...

**Comment by fgalan** [comment]
What do you mean by "all nodes" ?

**Comment by kzangeli** [comment]
With node I mean a VM or a physical machine where we run the functional tests

**Comment by kzangeli** [comment]
It would definitely be better if the headers came out in the same order that the broker sends them.
But, if this is not possible, and if yjr order now is to be random, the next best solution is for the accumulator to order them in a way that all nodes see them in the same order.

After that we'd have to touch each of our >1000 functests ... :(

The only good solution seems to be to try to maintain the order that the broker gives

**Comment by fgalan** [comment]
> The only good solution seems to be to try to maintain the order that the broker gives

I agree. And it seems to be doable, according to the SOF link I provide in the issue body. All nodes wuld need to use Python 3.7+ but I think is not a problem with virtualenv.
